File: GXpert/GXpert.Web/Modules/ServerTypes/Schools/SchoolColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SchoolRow } from "./SchoolRow";

export interface SchoolColumns {
    Id: Column<SchoolRow>;
    Name: Column<SchoolRow>;
    Description: Column<SchoolRow>;
    EType: Column<SchoolRow>;
    Address: Column<SchoolRow>;
    City: Column<SchoolRow>;
    StateTitle: Column<SchoolRow>;
    DistrictTitle: Column<SchoolRow>;
    TalukaTitle: Column<SchoolRow>;
    LocationInfo: Column<SchoolRow>;
    EstablishmentDate: Column<SchoolRow>;
    InsertDate: Column<SchoolRow>;
    InsertUserId: Column<SchoolRow>;
    UpdateDate: Column<SchoolRow>;
    UpdateUserId: Column<SchoolRow>;
    IsActive: Column<SchoolRow>;
}

export class SchoolColumns extends ColumnsBase<SchoolRow> {
    static readonly columnsKey = 'Schools.School';
    static readonly Fields = fieldsProxy<SchoolColumns>();
}

File: GXpert/GXpert.Web/Modules/Schools/School/SchoolColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Schools.Columns;

[ColumnsScript("Schools.School")]
[BasedOnRow(typeof(SchoolRow), CheckNames = true)]
public class SchoolColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public string EType { get; set; }
    public string Address { get; set; }
    public string City { get; set; }
    public string StateTitle { get; set; }
    public string DistrictTitle { get; set; }
    public string TalukaTitle { get; set; }
    public string LocationInfo { get; set; }
    public DateTime EstablishmentDate { get; set; }
}